packages/fluence-js/src/internal/FluencePeer.ts

```diff
@@ -615,14 +599,37 @@ export class FluencePeer {
                     // MUST happen sequentially (in a critical section).
                     // Otherwise the race between runner might occur corrupting the prevData
 
-                    const result = await runAvmRunner(status.peerId, this._avmRunner, item.particle, prevData);
-                    const newData = Buffer.from(result.data);
-                    prevData = newData;
+                    const args = serializeAvmArgs(
+                        {
+                            initPeerId: item.particle.initPeerId,
+                            currentPeerId: status.peerId,
+                            timestamp: item.particle.timestamp,
+                            ttl: item.particle.ttl,
+                        },
+                        item.particle.script,
+                        prevData,
+                        item.particle.data,
+                        item.particle.callResults,
+                    );
+
+                    item.particle.logTo('debug', 'Sending particle to interpreter');
+                    log.debug('prevData: ', dataToString(prevData));
+                    let avmCallResult: InterpreterResult | Error;
+                    try {
+                        const res = await this._fluenceAppService.callService('avm', 'invoke', args, undefined);
+                        avmCallResult = deserializeAvmResult(res);
+                    } catch (e) {
+                        avmCallResult = e instanceof Error ? e : new Error((e as any).toString());
+                    }
```

Comment:
not critical of course, but I think there is no need for a separate newData variable

Reply:
This was left intentionally to show what is being swapped with what. Otherwise is hard to track what is `data`, `prevdata` and `newdata`

Reply:
yeah. kind of agree